### PR TITLE
DOC: added string processing comparison with R

### PR DESCRIPTION
**Pull request by natethedrummer**
#13229

Added string processing section to comparison_with_r DOC. Used info from http://blog.dominodatalab.com/using-r-and-python-for-common-sas-functions.

This completes the issue.

### CODE REVIEW

File: doc/source/comparison_with_r.rst [hunk 1]
```diff
@@ -530,6 +530,103 @@ For more details and examples see :ref:`categorical introduction <categorical>`
 :ref:`differences to R's factor <categorical.rfactor>`.
```
Comment (jreback) [inline doc/source/comparison_with_r.rst]: add a section tag here, like: ``_compare_with_r.string`` (actually if you can add them some of the sub-sections would be great). you put right *after* the sub-section label.

File: doc/source/comparison_with_r.rst [hunk 2]
```diff
@@ -530,6 +530,103 @@ For more details and examples see :ref:`categorical introduction <categorical>`
...
+-----------------
+
+Length
+~~~~~~
+
+R determines the length of a character string with the ``nchar`` function. 
+``nchar`` includes leading and trailing blanks.  Use ``nchar`` and ``trimws`` 
+to exclude leading and trailing blanks. 
+
+.. code-block:: none
```
Comment (jreback) [inline doc/source/comparison_with_r.rst]: is there a R highlter?

File: doc/source/comparison_with_r.rst [hunk 3]
```diff
@@ -530,6 +530,103 @@ For more details and examples see :ref:`categorical introduction <categorical>`
...
+
+   df <- data.frame(color = c('red', ' blue', 'green ', ' yellow '))
+   nchar(as.character(df$color))
+   nchar(trimws(as.character(df$color)))
+
+Python determines the length of a character string with the ``len`` function.
+``len`` includes leading and trailing blanks.  Use ``len`` and ``strip`` 
+to exclude leading and trailing blanks.
+
+.. code-block:: none
```
Comment (jreback) [inline doc/source/comparison_with_r.rst]: -> ipython:: python (and for all running pandas code)

File: doc/source/comparison_with_r.rst [hunk 4]
```diff
@@ -530,6 +530,103 @@ For more details and examples see :ref:`categorical introduction <categorical>`
...
+
+R determines the length of a character string with the ``nchar`` function. 
+``nchar`` includes leading and trailing blanks.  Use ``nchar`` and ``trimws`` 
+to exclude leading and trailing blanks. 
+
+.. code-block:: none
+
+   df <- data.frame(color = c('red', ' blue', 'green ', ' yellow '))
+   nchar(as.character(df$color))
+   nchar(trimws(as.character(df$color)))
```
Comment (jreback) [inline doc/source/comparison_with_r.rst]: would be nice to show output here (IOW run the R code and show the output too if you can)

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/comparison_with_r.rst (hunk 2 above)
Comment (TomAugspurger) [inline doc/source/comparison_with_r.rst]: http://pygments.org/docs/lexers/#lexers-for-the-r-s-languages

`r` or `rconsole` should work.

Comment (jreback) [inline doc/source/comparison_with_r.rst]: yeah I realize that these produce the same output, so we don't actually show the output (I think) elsewhere, so maybe that is ok (though obviously the code formatting would be nice)

File: doc/source/comparison_with_r.rst (hunk 1 above)
Comment (TomAugspurger) [inline doc/source/comparison_with_r.rst]: If you aren't familiar with sphinx, you need to start the line with a `..` http://www.sphinx-doc.org/en/stable/markup/inline.html#cross-referencing-arbitrary-locations

**Comment by jreback** [comment]
@gfyoung can you review

File: doc/source/comparison_with_r.rst [hunk 5]
```diff
@@ -530,6 +530,103 @@ For more details and examples see :ref:`categorical introduction <categorical>`
...
+
+   df <- data.frame(sex = c('MALE', 'FEMALE'))
+   pos = regexpr("ALE", df$sex)
+   pos[1:2]
+
+Python determines the position of a character in a string with the 
+``find`` function.  ``find`` searches for the first position of the 
+substring.  If the substring is found, the function returns its 
+position.  Keep in mind that Python indexes are zero based whereas 
+R indexes are 1 based.
```
Comment (gfyoung) [inline doc/source/comparison_with_r.rst]: Simpler: "Keep in mind that Python 0-indexes, whereas R 1-indexes"

File: doc/source/comparison_with_r.rst [hunk 6]
```diff
@@ -530,6 +530,103 @@ For more details and examples see :ref:`categorical introduction <categorical>`
...
+with the ``substr`` function. 
+
+.. code-block:: none
+
+   df <- data.frame(sex = c('MALE', 'FEMALE'))
+   substr(df$sex, 1, 1)
+
+In Python, you can use ``[]`` notation to extract a substring 
+from a string by position locations.  Keep in mind that Python 
+indexes are zero-based.
```
Comment (gfyoung) [inline doc/source/comparison_with_r.rst]: Slightly simpler: "Keep in mind that Python 0-indexes"

File: doc/source/comparison_with_r.rst [hunk 7]
```diff
@@ -530,6 +530,103 @@ For more details and examples see :ref:`categorical introduction <categorical>`
...
+
+   df <- data.frame(name = c('Johnny Bravo', 'Alex Mack'))
+   toupper(df$name)
+   tolower(df$name)
+
+The equivalent Python functions are ``upper`` and ``lower``.  
+In addition, Python's ``title`` function changes the string to 
+proper case.
+
+.. code-block:: none
```
Comment (gfyoung) [inline doc/source/comparison_with_r.rst]: Same comment <a href="https://github.com/pandas-dev/pandas/pull/16502#discussion_r118583818">above</a>

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
can you update according to comments

**Comment by jreback** [comment]
closing as stale. if you'd like to continue working, pls ping.

**Pull request closed without merging**